escaped values on their own line cause erroneous compilation error

Somewhere between semanticdb version 4.7.8 and 4.8.15, the following bug was introduced. It is also reproducible in 4.9.1.

scala version: 2.13.12

# Bug reproduction steps
Attempt to compile the following code with `scalac Test.scala -Xplugin:semanticdb-plugin_singlejar.jar -Xsource:3`
```scala
object Test {
  val `f-oo` = 1
  def bar =
    `f-oo`
}
```
Note that it fails with
```
Test.scala:3: error: illegal start of simple expression
  def bar =
           ^
```
If the newline is removed
```scala
object Test {
  val `f-oo` = 1
  def bar = `f-oo`
}
```
it compiles just fine. If `-Xsource:3` or `-Xplugin:semanticdb-plugin_singlejar.jar` are left off, it compiles fine.

# Expected behavior
The semanticdb plugin does not cause compiler errors when run with `-Xsource:3` on code that compiles without the plugin

--- a/tests/shared/src/test/scala/scala/meta/tests/parsers/DefnSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/parsers/DefnSuite.scala
@@ -399,6 +399,30 @@ class DefnSuite extends ParseSuite {
     runTestAssert[Stat](code, layout)(tree)
   }
 
+  test("#3617 scala213") {
+    val codeWithoutBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |}
+         |""".stripMargin
+    val codeWithBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |
+         |}
+         |""".stripMargin
+    val layout = "object Test { def bar = `f-oo` }"
+    val tree = Defn.Object(
+      Nil,
+      tname("Test"),
+      tpl(Defn.Def(Nil, tname("bar"), Nil, None, tname("f-oo")))
+    )
+    runTestAssert[Stat](codeWithoutBlank, layout)(tree)
+    runTestAssert[Stat](codeWithBlank, layout)(tree)
+  }
+
   test("#3571 scala213source3") {
     implicit val Scala213 = scala.meta.dialects.Scala213Source3
     val code =
@@ -438,6 +462,31 @@ class DefnSuite extends ParseSuite {
       )
     )
     runTestAssert[Stat](code, layout)(tree)
+  }
+
+  test("#3617 scala213source3") {
+    implicit val Scala213 = scala.meta.dialects.Scala213Source3
+    val codeWithoutBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |}
+         |""".stripMargin
+    val codeWithBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |
+         |}
+         |""".stripMargin
+    val layout = "object Test { def bar = `f-oo` }"
+    val tree = Defn.Object(
+      Nil,
+      tname("Test"),
+      tpl(Defn.Def(Nil, tname("bar"), Nil, None, tname("f-oo")))
+    )
+    runTestAssert[Stat](codeWithoutBlank, layout)(tree)
+    runTestAssert[Stat](codeWithBlank, layout)(tree)
   }
 
 }

--- a/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/SignificantIndentationSuite.scala
+++ b/tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/SignificantIndentationSuite.scala
@@ -3381,4 +3381,28 @@ class SignificantIndentationSuite extends BaseDottySuite {
     runTestAssert[Stat](code, layout)(tree)
   }
 
+  test("#3617 scala3") {
+    val codeWithoutBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |}
+         |""".stripMargin
+    val codeWithBlank =
+      """|object Test {
+         |  def bar =
+         |    `f-oo`
+         |
+         |}
+         |""".stripMargin
+    val layout = "object Test { def bar = `f-oo` }"
+    val tree = Defn.Object(
+      Nil,
+      tname("Test"),
+      tpl(Defn.Def(Nil, tname("bar"), Nil, None, tname("f-oo")))
+    )
+    runTestAssert[Stat](codeWithoutBlank, layout)(tree)
+    runTestAssert[Stat](codeWithBlank, layout)(tree)
+  }
+
 }

